tutorial/write-custom-widget.md

```diff
@@ -107,3 +107,178 @@ Take care not to keep any important state in your renderer - animation tickers
 are well suited to that location but user state would not be. A widget that is
 hidden may have it's renderer destroyed and if it is shown again the new renderer
 must be able to reflect the same widget state.
+
+### Example Widget
+
+The following covers in detail the creation of a simple Widget and it's Renderer.
+
+#### Create a Widget
+
+So the widget requires the`widget.BaseWidget`. For example:
```

Comment:
Could it be better introduced? Dropping right in with "So" seems unclear.
Also there is a space missing